### PR TITLE
feat(da): move blob verification to the da service

### PR DESCRIPTION
**Pull request by itsdevbear**
## Summary by CodeRabbit

- **New Features**
	- Introduced a `DA (Data Availability)` service to manage blob sidecars processing in an event-driven architecture.
	- Added three new constants related to blob sidecars events: "blob-sidecars-built", "blob-sidecars-received", and "blob-sidecars-processed."

- **Refactor**
	- Renamed `ProcessBlockAndBlobs` to `ProcessBeaconBlock`, streamlining implementation for efficiency.
	- Consolidated parameters in multiple functions for improved readability and maintainability.

- **Dependencies**
	- Added new dependency on `github.com/berachain/beacon-kit/mod/async`.

- **Style**
	- Reformatted test cases and function invocations for better readability.

- **Tests**
	- Updated test cases in `genesis_test.go` for improved assertion structure.

### CODE REVIEW

**Review by archbear — APPROVED** [review]
lgtm i think pulling out blob processing makes a lot of sense

File: mod/da/pkg/da/service.go [hunk 1]
```diff
@@ -0,0 +1,172 @@
...
+		IsNil() bool
+	},
+	ExecutionPayloadT any,
+] struct {
+	avs AvailabilityStoreT
+	bp  BlobProcessor[
+		AvailabilityStoreT, BeaconBlockBodyT,
+		BlobSidecarsT, ExecutionPayloadT,
+	]
+	feed   *event.FeedOf[asynctypes.EventID, *asynctypes.Event[BlobSidecarsT]]
```
Comment (archbear) [inline mod/da/pkg/da/service.go:47]: feed + event type should be generics imo

Comment (itsdevbear) [reply mod/da/pkg/da/service.go:47]: yeah agreed, lemme remove and add in another PR then

Comment (itsdevbear) [reply mod/da/pkg/da/service.go]: we have some other spots in the repo we will have to cleanup as well

**Pull request merged**